core/src/HIP/Kokkos_HIP_Instance.hpp

```diff
@@ -144,6 +147,50 @@ class HIPInternal {
   void release_team_scratch_space(int scratch_pool_id);
 };
 
+template <typename DriverType>
+Kokkos::HIP::size_type *HIPInternal::scratch_functor(
+    DriverType const &driver) const {
```

Comment:
We should probably give it a better name now as it does not really capture that this function is copying the driver into a staging area in host pinned memory and returns a pointer to the device where the driver is being copied asynchronously.
Maybe a `transfer_to_device_and_get_pointer`?  Unless someone has a better idea because I am not super proud about that name...

Reply:
I agree that the current name is not great but I am not sure `transfer_to_device_and_get_pointer` is better. Unfortunately, I don't have a good name for this functions :-/

Reply:
`stage_functor` or `stage_functor_for_execution` ?